src/storage/src/source/mysql/snapshot.rs

```diff
@@ -0,0 +1,281 @@
+                    .query_drop(format!("LOCK TABLES {lock_clauses}"))
+                    .await?;
+
+                // Record the GTID set at the start of the snapshot
+                let snapshot_gtid_set = query_sys_var(lock_conn.as_mut().expect("lock_conn just created"), "global.gtid_executed").await?;
+                let snapshot_gtid_set = GtidSet::from_str(snapshot_gtid_set.as_str())?;
+                trace!(%id, "timely-{worker_id} acquired table locks at start gtid set: {snapshot_gtid_set:?}");
+                // TODO(roshan): Insert metric for how long it took to acquire the locks
+
+                // Send a signal to all workers that they should start their transactions
+                lock_start_handle.give(&lock_start_cap_set[0], snapshot_gtid_set).await;
+            }
+            *lock_start_cap_set = CapabilitySet::new();
+
+            // This worker has no tables to snapshot.
+            if reader_snapshot_table_info.is_empty() {
+                // Send a signal to the leader that this worker is okay with releasing the table locks
+                transaction_start_handle.give(&transaction_start_cap_set[0], ()).await;
+                *transaction_start_cap_set = CapabilitySet::new();
+                return Ok(());
+            }
+
+            let mut conn = connection_config.connect(&task_name, &config.config.connection_context.ssh_tunnel_manager).await?;
+
+            // Verify the MySQL system settings are correct for consistent row-based replication using GTIDs
+            // TODO: Should these return DefiniteError instead of TransientError?
+            ensure_gtid_consistency(&mut conn).await?;
+            ensure_full_row_binlog_format(&mut conn).await?;
+            ensure_replication_commit_order(&mut conn).await?;
+
+            // Wait for the start_gtids from the leader, which indicate that the table locks have
+            // been acquired and we should start a transaction.
+            let snapshot_gtid_set = loop {
+                match lock_start_input.next().await {
+                    Some(AsyncEvent::Data(_, mut data)) => break data.pop().expect("Sent above"),
+                    Some(AsyncEvent::Progress(_)) => (),
+                    None => panic!("lock_start_input closed unexpectedly"),
+                }
+            };
+
+            trace!(%id, "timely-{worker_id} starting transaction with consistent snapshot at gtid set: {snapshot_gtid_set:?}");
+
+            // Start a transaction with REPEATABLE READ and 'CONSISTENT SNAPSHOT' semantics
+            // so we can read a consistent snapshot of the table at the specific GTID we read.
+            let mut tx_opts = TxOpts::default();
+            tx_opts
+                .with_isolation_level(IsolationLevel::RepeatableRead)
+                .with_consistent_snapshot(true)
+                .with_readonly(true);
+            conn.start_transaction(tx_opts).await?;
+            conn.query_drop("set @@session.time_zone = '+00:00'")
```

Comment:
What does this setting affect? Would be good to have a comment explaining it

Reply:
Adding a comment. The docs (https://dev.mysql.com/doc/refman/8.0/en/datetime.html) say:

> MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval. (This does not occur for other types such as DATETIME.) 

So setting this means that all timestamp values will be returned in UTC